### ISSUE TITLE
`new Response` looks not fully compatible with node on Astro@4.4.0

### ISSUE THREAD

**Issue by Tanguy-Magnaudet** [oven-sh/bun#8923]
### What version of Bun is running?

1.0.26

### What platform is your computer?

Linux 6.2.0-39-generic x86_64 unknown (docker container)

### What steps can reproduce the bug?

Astro made a change on `4.4.0` and now uses an `AsyncIterable` instead of a `ReadableStream`.

While it still work fine on Node, when you use Bun as a runtime, it output `[Object object]` in the browser instead of the desired HTML.
As Bun claim a compatibility with node, it should work the same way.

### What is the expected behavior?

It should behave exactly like it behave on nodejs.

### What do you see instead?

An object is used in place of a string somewhere, causing the server to output it as [Object object] in the response instead of and HTML rendered page.

### Additional information

You can reproduce the issue by installing a sample Astro project in version 4.4.0. 
The version bellow (2.3.7) works fine. 

Here is the patchnote : [https://github.com/withastro/astro/releases/tag/astro%404.4.0](https://github.com/withastro/astro/releases/tag/astro%404.4.0)

And the specific PR : [https://github.com/withastro/astro/pull/9614](https://github.com/withastro/astro/pull/9614)

**Comment by paperclover** [comment]
The fix is to make the Response constructor accept any arbitrary async iterable in place of a ReadableStream. I assume it works identical to a readable stream.
```ts
// Node.js allows passing in an AsyncIterable to the Response constructor.
```

**Comment by paperclover** [comment]
This has been fixed in #8941. I verified this works by migrating my personal website to Astro 4.4, as well as testing some demo sites.
